### PR TITLE
Update dev docs with clean script note

### LINKED ISSUE #1171: Update dev docs with clean script note
Elyra developers can run a clean script in order to clean their environment prior to builds. This can be useful when developers are switching between major Elyra versions, as raised in #1128.
Dev docs could be updated with this info.

### PR DESCRIPTION
**Pull request by karlaspuldaro**
Fixes  #1171 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: docs/source/developer_guide/development-workflow.md [hunk 1]
```diff
@@ -68,6 +68,16 @@ You can check that the JupyterLab extension was successful installed with:
 jupyter labextension list
 ```
 
+> NOTE: 
+When switching between Elyra major versions, it is recommended to clean your JupyterLab environment before a build.
+The `clean-jupyterlab` uninstalls your JupyterLab packages and completely clean/reinstall your Jupyter workspace.
+Make sure to save any important data in your environment before running the script.
+Please refer to the disclaimer at the top of the script file.
```
Comment (ptitzler) [inline docs/source/developer_guide/development-workflow.md:75]: How about echo-ing the disclaimer and forcing the user to confirm when the script is run, like so in `clean-jupyterlab.sh `

```
echo "This script removes your JupyterLab installation ...."
echo "..."
read -p "Hit ENTER to continue or CTRL C to abort"
```

This way all information is in one place.
If there's a use case scenario where the script is part of an automated process (is there?) the last line of code would need to be guarded.

Comment (karlaspuldaro) [reply docs/source/developer_guide/development-workflow.md:75]: I like this approach. 
And afaik it's not automated. @lresende @akchinSTC can confirm.

File: etc/scripts/clean-jupyterlab.sh [hunk 2]
```diff
@@ -14,6 +14,26 @@
...
 #
+################################################################################
+# DISCLAIMER:
+# *** This script will completely clean your Jupyter/JupyterLab environment. ***
+# *** Any important data stored in your Jupyter/JupyterLab workspace will be removed. ***
+# *** Any package installed in your Jupyter/JupyterLab workspace will be removed. ***
+#
+#
+# INSTRUCTIONS:
+# This script is intended to help setting up a clean environment for Elyra's development.
```
[the remark this answers is not in the document]
Comment (lresende) [reply etc/scripts/clean-jupyterlab.sh:25]: Also maybe add the locations that will be deleted

File: docs/source/developer_guide/development-workflow.md (hunk 1 above)
Comment (lresende) [reply docs/source/developer_guide/development-workflow.md:75]: what do you mean by `it's not automated`?

Comment (ptitzler) [reply docs/source/developer_guide/development-workflow.md:75]: for example, the script is not invoked as part of any test or in any way that currently does not require any user interaction to complete

**Review by ajbozarth — APPROVED** [review]

**Review by lresende — APPROVED** [review]
+0

**Review by marthacryan — APPROVED** [review]

**Review by akchinSTC — APPROVED** [review]
LGTM

**Pull request merged**